test/e2e/set/scyllacluster/scyllacluster_expose.go

```diff
@@ -5,25 +5,248 @@ package scyllacluster
+
+				broadcastRPCAddress, err := configClient.BroadcastRPCAddress(ctx)
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(broadcastRPCAddress).To(o.Equal(svc.Spec.ClusterIP))
+			},
+		}),
+		g.Entry("using ClusterIP Service, and broadcast it to nodes, and PodIP from Pod Status to clients", &entry{
+			exposeOptions: &scyllav1.ExposeOptions{
+				NodeService: &scyllav1.NodeServiceTemplate{
+					Type: scyllav1.ClusterIPServiceType,
+				},
+				BroadcastOptions: &scyllav1.NodeBroadcastOptions{
+					Nodes: &scyllav1.BroadcastOptions{
+						Type: scyllav1.ServiceClusterIPBroadcastAddressType,
+					},
+					Clients: &scyllav1.BroadcastOptions{
+						Type: scyllav1.PodIPBroadcastAddressType,
+						PodIP: &scyllav1.PodIPAddressOptions{
+							Source: scyllav1.StatusPodIPSource,
+						},
+					},
+				},
+			},
+			validateService: func(ctx context.Context, client corev1client.CoreV1Interface, sc *scyllav1.ScyllaCluster) {
+				serviceList, err := f.KubeClient().CoreV1().Services(sc.Namespace).List(ctx, metav1.ListOptions{
+					LabelSelector: utils.GetMemberServiceSelector(sc.Name).String(),
+				})
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				svc := serviceList.Items[0]
+				o.Expect(svc.Spec.Type).To(o.Equal(corev1.ServiceTypeClusterIP))
+				parsedClusterIP := net.ParseIP(svc.Spec.ClusterIP)
+				o.Expect(parsedClusterIP).ToNot(o.BeNil())
+			},
+			validateScyllaConfig: func(ctx context.Context, configClient *scyllaclient.ConfigClient, svc *corev1.Service, pod *corev1.Pod) {
+				broadcastAddress, err := configClient.BroadcastAddress(ctx)
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				o.Expect(broadcastAddress).To(o.Equal(svc.Spec.ClusterIP))
+
+				broadcastRPCAddress, err := configClient.BroadcastRPCAddress(ctx)
+				o.Expect(err).NotTo(o.HaveOccurred())
+				o.Expect(broadcastRPCAddress).To(o.Equal(pod.Status.PodIP))
+			},
+		}),
+		g.Entry("using Headless Service, and broadcast PodIP from Pod Status to both client and nodes", &entry{
+			exposeOptions: &scyllav1.ExposeOptions{
+				NodeService: &scyllav1.NodeServiceTemplate{
+					Type: scyllav1.HeadlessServiceType,
+				},
+				BroadcastOptions: &scyllav1.NodeBroadcastOptions{
+					Nodes: &scyllav1.BroadcastOptions{
+						Type: scyllav1.PodIPBroadcastAddressType,
+						PodIP: &scyllav1.PodIPAddressOptions{
+							Source: scyllav1.StatusPodIPSource,
+						},
+					},
+					Clients: &scyllav1.BroadcastOptions{
+						Type: scyllav1.PodIPBroadcastAddressType,
+						PodIP: &scyllav1.PodIPAddressOptions{
+							Source: scyllav1.StatusPodIPSource,
+						},
+					},
+				},
+			},
+			validateService: func(ctx context.Context, client corev1client.CoreV1Interface, sc *scyllav1.ScyllaCluster) {
+				serviceList, err := f.KubeClient().CoreV1().Services(sc.Namespace).List(ctx, metav1.ListOptions{
+					LabelSelector: utils.GetMemberServiceSelector(sc.Name).String(),
+				})
+				o.Expect(err).NotTo(o.HaveOccurred())
+
+				svc := serviceList.Items[0]
```

Comment:
you're listing the services in every instance of `validateService` - can't you just get the service in Describe and pass it to the function as a parameter?

Reply:
done